cmd/environment.go

```diff
@@ -0,0 +1,148 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ory/viper"
+	"github.com/spf13/cobra"
+
+	"knative.dev/func/pkg/config"
+	"knative.dev/func/pkg/functions"
+	"knative.dev/func/pkg/k8s"
+)
+
+func NewEnvironmentCmd(newClient ClientFactory, version *Version) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "environment",
+		Short: "Display function execution environment information",
+		Long: `
+NAME
+	{{rootCmdUse}} environment
+
+SYNOPSIS
+	{{rootCmdUse}} environment - display function execution environment information
+
+DESCRIPTION
+	Display information about the function execution environment, including
+	the version of func, the version of the function spec, the default builder,
+	available runtimes, and available templates.
+`,
+		SuggestFor: []string{"env", "environemtn", "enviroment", "enviornment", "enviroment"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runEnvironment(cmd, newClient, version)
+		},
+	}
+	cfg, err := config.NewDefault()
+	if err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "error loading config at '%v'. %v\n", config.File(), err)
+	}
+
+	addVerboseFlag(cmd, cfg.Verbose)
+
+	return cmd
+}
+
+type Environment struct {
+	Version     string
+	Build       string
+	SpecVersion string
+	SocatImage  string
+	TarImage    string
+	Languages   []string
+	Templates   map[string][]string
+	Environment []string
+	Cluster     string
+	Defaults    config.Global
+}
+
+func runEnvironment(cmd *cobra.Command, newClient ClientFactory, v *Version) (err error) {
+	cfg, err := newEnvironmentConfig()
+	if err != nil {
+		return
+	}
+
+	client := functions.New(functions.WithVerbose(cfg.Verbose))
+	r, err := getRuntimes(client)
+	if err != nil {
+		return
+	}
+	t, err := getTemplates(client, r)
+	if err != nil {
+		return
+	}
+	var envs []string
+	for _, e := range os.Environ() {
+		if strings.HasPrefix(e, "FUNC_") {
+			envs = append(envs, e)
+		}
+	}
+	defaults, err := config.NewDefault()
+	if err != nil {
+		return
+	}
+
+	var host string
+	cc, err := k8s.GetClientConfig().ClientConfig()
+	if err != nil {
+		fmt.Printf("Error getting client config %v\n", err)
+	} else {
+		host = cc.Host
+	}
+
+	environment := Environment{
+		Version:     v.String(),
+		Build:       v.Hash,
```

Comment:
I would call this field `Git Revision`.
Also I would add `Build Date` field which is more telling to normal user.

Reply:
```diff
[k8s:kind@kind-func/default] (venv) ~/devel/kn-plugin-func   func-environment ±  git diff | cat                                                                                                              ST 100   func-environment 
diff --git a/cmd/environment.go b/cmd/environment.go
index 8e63bad8..b987f13c 100644
--- a/cmd/environment.go
+++ b/cmd/environment.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"os"
 	"strings"
+	"time"
 
 	"github.com/ory/viper"
 	"github.com/spf13/cobra"
@@ -47,7 +48,8 @@ DESCRIPTION
 
 type Environment struct {
 	Version     string
-	Build       string
+	GitRevision string
+	BuildDate   string
 	SpecVersion string
 	SocatImage  string
 	TarImage    string
@@ -92,9 +94,15 @@ func runEnvironment(cmd *cobra.Command, newClient ClientFactory, v *Version) (er
 		host = cc.Host
 	}
 
+	d, err := time.Parse("20060102T150405Z", v.Date)
+	if err != nil {
+		panic(err)
+	}
+
 	environment := Environment{
 		Version:     v.String(),
-		Build:       v.Hash,
+		GitRevision: v.Hash,
+		BuildDate:   d.Local().Format(time.DateTime),
 		SpecVersion: functions.LastSpecVersion(),
 		SocatImage:  k8s.SocatImage,
 		TarImage:    k8s.TarImage,
```